components/config/src/config/mod.rs

```diff
@@ -262,26 +393,39 @@ impl Config {
         self.highlight_code = false;
     }
 
-    pub fn get_translation<S: AsRef<str>>(&self, lang: S, key: S) -> Result<String> {
-        let terms = self.translations.get(lang.as_ref()).ok_or_else(|| {
-            Error::msg(format!("Translation for language '{}' is missing", lang.as_ref()))
-        })?;
+    pub fn get_localized(&self, lang: &LanguageIdentifier) -> Result<LocalizedConfig> {
+        if lang == &self.default_language {
+            Ok(LocalizedConfig(self.clone()))
+        } else if self.languages.contains_key(lang) {
+            let mut languages = self.languages.clone();
+            let (_, default_language_options) = languages.remove_entry(&lang).unwrap();
+            languages.insert(self.default_language.clone(), self.default_language_options.clone());
+            Ok(LocalizedConfig(Config {
+                lang: lang.clone(),
+                default_language_options,
+                languages,
+                ..self.clone()
```

Comment:
I guess this means we clone the Config every time we render a template?

Reply:
Unfortunately, yes. I'm thinking about having a struct with just references like `SerializingTaxonomy` and co. I'm way too inexperienced to not resort to cloning something with that many borrows.